### PR TITLE
feat: Improve the response format

### LINKED ISSUE #8: [FEATURE] Improve Response Format
### Description

The search response it is generating is coming in markdown format. I will make some changes in the prompt to make more readable and correctly formatted. 

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by itsBaivab**
### 🛠️ Fixes Issue

Closes #8 

### 👨‍💻 Changes proposed

Improve the response format (Used Regex to post process the markdown in Plain text format tried to cover all the markdown scenarios.)

### ✔️ Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Review by Pradumnasaraf — COMMENTED** [review]
Hey,

Thanks for the PR. I was thinking that instead of using regexp, we could mention in the prompt that we want the output ~~with~~ without horizontal rules, italics, etc. What do you think?

**Comment by Pradumnasaraf** [comment]
> In my opinion for all kind of question hard coding a single answer format is not going to look good. Additionally, before using regex I was tiring to do some prompt manipulation to get our desired answer format, but the results are not consistent throughout each and every question.
> 
> also tried multiple markdown rendering modules but is tremendously affecting the response speed by 5..6 sec.

I agree, that can be painful! Thank you for your thoughts :)

File: cmd/searchCmd.go [hunk 1]
```diff
@@ -47,7 +46,31 @@ func getApiRespone(args []string) string {
...
+	re = regexp.MustCompile(`\n---\n`)
+	input = re.ReplaceAllString(input, "\n")
+
+	// Remove italic formatting
+	re = regexp.MustCompile(`_([^_]+)_`)
+	input = re.ReplaceAllString(input, "$1")
+
+	// Remove bold formatting
+	re = regexp.MustCompile(`\*\*([^*]+)\*\*`)
+	input = re.ReplaceAllString(input, "$1")
```
Comment (Pradumnasaraf) [inline cmd/searchCmd.go:67]:
```suggestion
	// Remove italic formatting
	re = regexp.MustCompile(`_([^_]+)_|\*([^*]+)\*`)
	input = re.ReplaceAllString(input, "$1$2")

	// Remove bold formatting
	re = regexp.MustCompile(`\*\*([^*]+)\*\*|__([^_]+)__`)
	input = re.ReplaceAllString(input, "$1$2")
```

**Review by Pradumnasaraf — CHANGES_REQUESTED** [review]
Hey,

Thanks for the PR. Inclusing some edge cases

**Comment by Pradumnasaraf** [comment]
Seems like it's working correctly as well:

Screenshot 2024-07-22 at 4 14 56 PM

**Comment by itsBaivab** [comment]
> Seems like it's working correctly as well:
> 
> Screenshot 2024-07-22 at 4 14 56 PM

is it still generating bold formating?

**Comment by itsBaivab** [comment]
> > Seems like it's working correctly as well:
> > Screenshot 2024-07-22 at 4 14 56 PM
> 
> is it still generating bold formatting?

image


This is the response I am getting

**Review by Pradumnasaraf — APPROVED** [review]
Let's merge it. Improvements can come later in the other PRs.

Thank you so much!

**Pull request merged**